WireUI/Sources/WireIndividualToTeamMigrationUI/IndividualToTeamMigrationViewController.swift

```diff
@@ -0,0 +1,325 @@
+                    key: "individualToTeam.teamName.closeButton.accessibilityLabel",
+                    bundle: .module
+                )
+            case .confirmation:
+                .localizedAccessibilityLabel(
+                    key: "individualToTeam.confirmation.closeButton.accessibilityLabel",
+                    bundle: .module
+                )
+            case .completion:
+                .localizedAccessibilityLabel(
+                    key: "individualToTeam.completion.closeButton.accessibilityLabel",
+                    bundle: .module
+                )
+            }
+        }
+
+        private var titleStringKey: String.LocalizationValue {
+            switch self {
+            case .teamPlanSelection:
+                "individualToTeam.planSelection.title"
+            case .teamName:
+                "individualToTeam.teamName.title"
+            case .confirmation:
+                "individualToTeam.confirmation.title"
+            case .completion:
+                "individualToTeam.completion.title"
+            }
+        }
+    }
+
+    enum Transition: Sendable {
+        case toCancellationAlert
+        case dismissCancellationAlert
+        case toPlans
+        case toTeamName
+        case toConfirmation(teamName: String)
+        case toTeamCreation(teamName: String)
+        case toError(error: any Error)
+        case toCompletion(teamName: String)
+        case toApp
+        case toTeamManagement
+    }
+
+    let actionCallback: @Sendable (Action) -> Void
+    lazy var blockingActivityIndicator: BlockingActivityIndicator = BlockingActivityIndicator(
+        view: view,
+        accessibilityAnnouncement: .localizedAccessibilityLabel(key: "loading", bundle: .module)
+    )
+    let childController: UINavigationController
+    var currentStep: Step
+    let features: [TeamPlanFeature]
+    let useCase: any IndividualToTeamMigrationUseCase
+    let userProfileName: String
+
+    public init(
+        features: [TeamPlanFeature],
+        useCase: any IndividualToTeamMigrationUseCase,
+        userProfileName: String,
+        actionCallback: @escaping @Sendable (Action) -> Void
+    ) {
+        self.actionCallback = actionCallback
+        self.currentStep = .teamPlanSelection(features: features)
+        self.childController = UINavigationController()
+        self.features = features
+        self.useCase = useCase
+        self.userProfileName = userProfileName
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    @available(*, unavailable)
+    required init?(coder: NSCoder) {
+        fatalError()
+    }
+
+    public override func viewDidLoad() {
+        super.viewDidLoad()
+        addChild(childController)
+        view.addSubview(childController.view)
+        childController.didMove(toParent: self)
+        childController.navigationBar.tintColor = .darkText
+        transition(to: .toPlans)
+    }
+
+    @MainActor
+    func transition(to transition: Transition) {
```

Comment:
What is the difference/advantage of having a navigation controller as child over not having a custom container view controller at all?

Reply:
What are you suggesting instead?

Reply:
So the container is basically just for routing purposes?
I think a router should be a simple type which deals with the (plain, not subclassed) navigation controller.
Let's keep it for now.

Reply:
Agreed. But then, in this context, who owns and keeps a reference to, the router?
The main app?

Reply:
Yeah, I see how that would work. But now the views would be aware of the router.

Ideally, there would be a parent router, and we wouldn’t have to dance around with references. But that will come.